xregexp-all.js

```diff
@@ -298,7 +300,7 @@ module.exports = function(XRegExp) {
             // Using `XRegExp.union` safely rewrites backreferences in `left` and `right`
             esc = new RegExp(
                 '(?:' + escapeChar + '[\\S\\s]|(?:(?!' +
-                    XRegExp.union([left, right]).source +
+                    XRegExp.union([left, right], basicFlags, {conjunction: 'or'}).source +
```

Comment:
It's not necessary and possibly inappropriate to add `basicFlags` here, since those flags were already applied to `left` and `right` when those strings were passed through the `XRegExp` constructor a few lines up. In most cases it wouldn't matter if you apply the flags again, but in edge cases with user-added custom flag addons that transform syntax that they also output, this could lead to transforming the original output in a way that isn't intended.

I noted it's "possibly" inappropriate, because honestly this edge case is unavoidable given the flexibly of user-added syntax transforms. They could be transforming their own output even in cases that don't rely on a flag. Hopefully no addons do that.

I'll go ahead and merge this though, since I can revert this part while adding a comment about it.